0005-crashtracker-structured-log-format.md

```diff
@@ -0,0 +1,203 @@
+    A `Thread` object has the following fields:
+    - `crashed`: **[required]**
+      A boolean which tells if the thread crashed.
+    - `name`: **[required]**
+      Name of the thread (e.g. 'Thread 0').
+    - `stack`: **[required]**
+      The `StackTrace` of the thread.
+      See below for more details on how stacktraces are formatted.
+    - `state`: **[optional]**
+      Platform-specific state of the thread when its state was captured (CPU registers dump for iOS, thread state enum for Android, etc.).
+      Currently, this is a platform-dependent string.
+  - `is_crash`: **[required]**
+    Boolean true if the error was a crash, false otherwise.
+  - `kind`: **[required]**
+    The kind of error that occurred.
+    For a crash, a non-exhaustive set of options include "SigAbort", "SigBus", "SigSegv".
+    This field MAY be extended to include options such at "Panic", "UnhandledException" etc.
+  - `message`: **[optional]**
+    A human readable string containing an error message associated with the stack trace.
+  - `source_type`: **[required]**
+    The string "crashtracking".
+  - `stack`: **[required]**
+    This represents the stack of the crashing thread.
+    See below for more details on how stacktraces are formatted.
+- `files`: **[optional]**
+  A `Map<filename, contents>` where `contents` is an array of plain text strings, one per line.
+  Useful files for triage and debugging, such as `/proc/self/maps` or `/proc/meminfo`.
+- `incomplete`: **[required]**
+  Boolean `false` if the crashreport is complete (i.e. contains all intended data), `true` if there is expected missing data.
+  This can happen becasue the crashtracker is architected to stream data to an out of process receiver, allowing a partial crash report to be emitted even in the case where the crashtracker itself crashed during stack trace collection.
+  This MUST be set to `true` if any required field is missing.
+- `metadata`: **[required]**
+  Metadata about the system in which the crash occurred:
+  - `library_name`: **[required]**
+    e.g. "dd-trace-python".
+  - `library_version`: **[required]**
+    e.g. "2.16.0".
+  - `family`: **[required]**
+    e.g. "python".
+  - `tags`: **[optional]**
+    A set of key:value pairs, representing any tags the crashtracking system wishes to associate with the crash.
+    Examples would include "hostname", "service", and any configuration information the system wishes to track.
+- `os_info`: **[required]**
+  The OS + processor architecture on which the crash occurred. TODO: define how these should look with examples.
+  - `architecture`: **[required]**
+  - `bitness`: **[required]**
+  - `os_type`: **[required]**
+  - `version`: **[required]**
+- `proc_info`: **[optional]**
+  A place to store information about the crashing process.
+  In the future, this may have additional optional fields as more data is collected.
+  - `pid`: **[required]**
+    The PID of the crashing process.
+- `sig_info`: **[optional]**
```

Comment:
As discussed, might be good to harmonize the names of these with the field names used in the siginfo_t.  In particular, we'll rename faulting_address -> si_addr

Reply:
done